filecoin-proofs/src/api/post.rs

```diff
@@ -604,18 +595,18 @@ pub fn verify_winning_post<Tree: 'static + MerkleTreeTrait>(
     let pub_params: compound_proof::PublicParams<fallback::FallbackPoSt<Tree>> =
         fallback::FallbackPoStCompound::setup(&setup_params)?;
 
-    let mut pub_sectors = Vec::with_capacity(param_sector_count);
-    for _ in 0..param_sector_count {
-        for (sector_id, replica) in replicas.iter() {
+    let pub_sectors: Vec<_> = replicas
+        .iter()
+        .map(|(sector_id, replica)| {
             let comm_r = replica.safe_comm_r().with_context(|| {
-                format!("verify_winning_post: safe_comm_r failed: {:?}", sector_id)
+                format!("verify_window_post: safe_comm_r failed: {:?}", sector_id)
             })?;
-            pub_sectors.push(fallback::PublicSector {
+            Ok(fallback::PublicSector {
                 id: *sector_id,
                 comm_r,
-            });
-        }
-    }
+            })
+        })
+        .collect::<Result<_>>()?;
```

Comment:
I think the problem with this change is that it assumed each sector could only appear once in the set of challenged sectors. But in the case of Winning PoSt, there is just one sector, yet it is challenged repeatedly (each time with only a single challenge).

Yes, this is strange and confusing. Yes, the pervasive naming makes this almost impossible to understand.